idemp-rest/WEB-INF/src/org/bandahealth/idempiere/rest/service/db/BaseInvoiceDBService.java

```diff
@@ -145,14 +147,22 @@ public T saveEntity(T entity) {
 			invoice.setIsApproved(true);
 			invoice.setDocAction(MInvoice_BH.DOCACTION_Complete);
 			invoice.setPaymentRule(entity.getPaymentRule());
+			invoice.setBH_InvoiceType(entity.getInvoiceType());
 
 			beforeSave(entity, invoice);
 
 			// set target document type
 			if (!invoice.isSOTrx()) {
-				int apInvoiceId = getAPInvoiceDocumentTypeId();
-				invoice.setC_DocType_ID(apInvoiceId);
-				invoice.setC_DocTypeTarget_ID(apInvoiceId);
+				// income should have ARI docType
+				int docTypeId;
+				if (MInvoice_BH.INCOME_InvoiceType == entity.getInvoiceType()) {
+					docTypeId = getInvoiceDocumentTypeId(MDocType.DOCBASETYPE_ARInvoice);
+				} else {
+					docTypeId = getInvoiceDocumentTypeId(MDocType.DOCBASETYPE_APInvoice);
+				}
+
+				invoice.setC_DocType_ID(docTypeId);
+				invoice.setC_DocTypeTarget_ID(docTypeId);
 			}
```

Comment:
Do we need an `else` block to handle sales invoices? Not sure, just asking.

Reply:
There is a default for orders set via a model event, but we don't have anything to handle invoices that are created without orders. I think a default will probably be set somewhere, but I'd love to be intentional with it.

Reply:
Yes, kindly check out the latest. It has an else block. Thanks